src/connectors/impls/gcl/writer.rs

```diff
@@ -0,0 +1,239 @@
+// Copyright 2022, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub(crate) mod meta;
+mod sink;
+
+use crate::connectors::impls::gcl::writer::sink::GclSink;
+use crate::connectors::prelude::*;
+use crate::connectors::{Connector, ConnectorBuilder, ConnectorConfig, ConnectorType};
+use crate::errors::Error;
+use googapis::google::api::MonitoredResource;
+use googapis::google::logging::r#type::LogSeverity;
+use serde::Deserialize;
+use simd_json::OwnedValue;
+use std::collections::HashMap;
+use tremor_pipeline::ConfigImpl;
+
+#[derive(Deserialize, Clone)]
+pub(crate) struct Config {
+    // Optional. A default log resource name that is assigned to all log entries in entries that do not specify a value for log_name:
+    //
+    // "projects/\[PROJECT_ID]/logs/[LOG_ID\]"
+    // "organizations/\[ORGANIZATION_ID]/logs/[LOG_ID\]"
+    // "billingAccounts/\[BILLING_ACCOUNT_ID]/logs/[LOG_ID\]"
+    // "folders/\[FOLDER_ID]/logs/[LOG_ID\]"
+    // \[LOG_ID\] must be URL-encoded. For example:
+    //
+    // "projects/my-project-id/logs/syslog"
+    // "organizations/1234567890/logs/cloudresourcemanager.googleapis.com%2Factivity"
+    // The permission logging.logEntries.create is needed on each project, organization, billing account, or folder that is receiving new log entries, whether the resource is specified in logName or in an individual log entry.
+    pub log_name: Option<String>,
+
+    // Optional. A default monitored resource object that is assigned to all log entries in entries that do not specify a value for resource. Example:
+    //
+    // { "type": "gce_instance",
+    // "labels": {
+    //   "zone": "us-central1-a", "instance_id": "00000000000000000000" }}
```

Comment:
```suggestion
    /// ```json
    /// { "type": "gce_instance",
    /// "labels": {
    ///   "zone": "us-central1-a", "instance_id": "00000000000000000000" }}
    /// ```    
```